src/conf_mode/protocols_nhrp.py

```diff
@@ -0,0 +1,139 @@
+    if config:
+        conf = config
+    else:
+        conf = Config()
+    base = ['protocols', 'nhrp']
+
+    nhrp = conf.get_config_dict(base, key_mangling=('-', '_'),
+                                get_first_key=True, no_tag_node_value_mangle=True)
+
+    diff = get_config_diff(conf, key_mangling=('-', '_'))
+    diff_nodes = diff.get_child_nodes_diff(base + ['tunnel'])
+    nhrp['new_tunnels'] = diff_nodes['add']
+    nhrp['del_tunnels'] = diff_nodes['delete']
+
+    if not conf.exists(base):
+        return nhrp
+
+    nhrp['if_tunnel'] = conf.get_config_dict(['interfaces', 'tunnel'], key_mangling=('-', '_'),
+                                get_first_key=True, no_tag_node_value_mangle=True)
+
+    nhrp['profile_map'] = {}
+    profile = conf.get_config_dict(['vpn', 'ipsec', 'profile'], key_mangling=('-', '_'),
+                                get_first_key=True, no_tag_node_value_mangle=True)
+
+    for name, profile_conf in profile.items():
+        if 'bind' in profile_conf and 'tunnel' in profile_conf['bind']:
+            interfaces = profile_conf['bind']['tunnel']
+            if isinstance(interfaces, str):
+                interfaces = [interfaces]
+            for interface in interfaces:
+                nhrp['profile_map'][interface] = name
+
+    return nhrp
+
+def verify(nhrp):
+    if 'tunnel' in nhrp:
+        for name, nhrp_conf in nhrp['tunnel'].items():
+            if not nhrp['if_tunnel'] or name not in nhrp['if_tunnel']:
+                raise ConfigError(f'Tunnel interface "{name}" does not exist')
+
+            tunnel_conf = nhrp['if_tunnel'][name]
+
+            if 'encapsulation' not in tunnel_conf or tunnel_conf['encapsulation'] != 'gre':
+                raise ConfigError(f'Tunnel "{name}" is not an mGRE tunnel')
+
+            if 'remote' in tunnel_conf:
+                raise ConfigError(f'Tunnel "{name}" cannot have a remote address defined')
+
+            if 'map' in nhrp_conf:
+                for map_name, map_conf in nhrp_conf['map'].items():
+                    if 'nbma_address' not in map_conf:
+                        raise ConfigError(f'nbma-address missing on map {map_name} on tunnel {name}')
+
+            if 'dynamic_map' in nhrp_conf:
+                for map_name, map_conf in nhrp_conf['dynamic_map'].items():
+                    if 'nbma_domain_name' not in map_conf:
+                        raise ConfigError(f'nbma-domain-name missing on dynamic-map {map_name} on tunnel {name}')
+    return None
+
+def generate(nhrp):
+    render('/etc/opennhrp/opennhrp.conf', 'nhrp/opennhrp.conf.tmpl', nhrp)
+    return None
+
+def apply_iptables(iface, tunnel_conf):
+    if not tunnel_conf or 'source_address' not in tunnel_conf:
+        return None
+
+    chain = f'VYOS_NHRP_{iface}_OUT_HOOK'
+    source_addr = tunnel_conf['source_address']
+    try:
+        cmd(f'sudo iptables -N {chain}')
+        cmd(f'sudo iptables -A {chain} -p gre -s {source_addr} -d 224.0.0.0/4 -j DROP')
```

Comment:
This could become a problem - I had the same in `conntrack.py`. If only minor parameters of the tunnel are adjusted, the iptables rule is installed multiple times, thus you should `iptables --check` if the rule probably already exists.

Please also use the long option names as it makes it easier to read.